tests/prague/eip7702_set_code_tx/test_set_code_txs_2.py

```diff
@@ -0,0 +1,1722 @@
+        + gas_costs.G_COLD_SLOAD  # storage access under pointer call is cold
+        + G_CALL_OPCODE
+        + opcodes_price
+    )
+    direct_call_gas = (
+        gas_costs.G_STORAGE_SET
+        + gas_costs.G_WARM_ACCOUNT_ACCESS  # since previous pointer call, contract is now warm
+        + gas_costs.G_COLD_SLOAD  # but storage is cold, because it's contract's direct
+        + G_CALL_OPCODE
+        + opcodes_price
+        - 2  # because direct call is cheaper?
+    )
+
+    contract = pre.deploy_contract(code=Op.SSTORE(call_worked, Op.ADD(Op.SLOAD(call_worked), 1)))
+
+    storage_test_gas: Storage = Storage()
+    contract_test_gas = pre.deploy_contract(
+        code=Op.MSTORE(1000, Op.GAS())
+        + Op.CALL(gas=100_000, address=pointer_a)
+        + Op.SSTORE(
+            storage_test_gas.store_next(pointer_call_gas, "pointer_call_price"),
+            Op.SUB(Op.MLOAD(1000), Op.GAS()),
+        )
+        + Op.MSTORE(2000, Op.GAS())
+        + Op.CALL(gas=100_000, address=contract)
+        + Op.SSTORE(
+            storage_test_gas.store_next(direct_call_gas, "direct_call_price"),
+            Op.SUB(Op.MLOAD(2000), Op.GAS()),
+        )
+    )
+
+    tx = Transaction(
+        to=contract_test_gas,
+        gas_limit=3_000_000,
+        data=b"",
+        value=0,
+        sender=sender,
+        authorization_list=(
+            [
+                AuthorizationTuple(
+                    address=contract,
+                    nonce=0,
+                    signer=pointer_a,
+                )
+            ]
+        ),
+    )
+
+    post = {
+        contract: Account(storage={call_worked: 1}),
+        pointer_a: Account(storage={call_worked: 1}),
+        contract_test_gas: Account(storage=storage_test_gas),
+    }
+    state_test(
+        env=env,
+        pre=pre,
+        post=post,
+        tx=tx,
+    )
+
+
+@pytest.mark.valid_from("Prague")
+def test_pointer_to_static(state_test: StateTestFiller, pre: Alloc):
+    """
+    Tx -> call -> pointer A -> static -> static violation
+    Verify that static context is active when called under pointer.
+    """
+    env = Environment()
+    storage: Storage = Storage()
+    sender = pre.fund_eoa()
+    pointer_a = pre.fund_eoa()
+
+    contract_b = pre.deploy_contract(code=Op.SSTORE(0, 5))
+    contract_a = pre.deploy_contract(
+        code=Op.SSTORE(
+            storage.store_next(0, "static_call"),
+            Op.STATICCALL(1_000_000, contract_b, 0, 32, 1000, 32),
+        )
+        + Op.SSTORE(storage.store_next(1, "call_worked"), 1)
+    )
+
+    tx = Transaction(
+        to=pointer_a,
+        gas_limit=3_000_000,
+        data=b"",
+        value=0,
+        sender=sender,
+        authorization_list=[
+            AuthorizationTuple(
+                address=contract_a,
+                nonce=0,
+                signer=pointer_a,
+            )
+        ],
+    )
+
+    post = {pointer_a: Account(storage=storage)}
```

Comment:
```suggestion
    post = {pointer_a: Account(storage=storage), contract_b: Account(storage={0: 0})}
```
Just to double check that the storage did not get modified.